lib/GetFilenameFromUrl.js

```diff
@@ -1,21 +1,36 @@
 var pathJoin = require("./PathJoin");
+var urlParse = require("url").parse;
 
 function getFilenameFromUrl(publicPath, outputPath, url) {
-	// publicPrefix is the folder our bundle should be in
-	var localPrefix = publicPath || "/";
-	if(url.indexOf(localPrefix) !== 0) {
-		if(/^(https?:)?\/\//.test(localPrefix)) {
-			localPrefix = "/" + localPrefix.replace(/^(https?:)?\/\/[^\/]+\//, "");
-			// fast exit if another directory requested
-			if(url.indexOf(localPrefix) !== 0) return false;
-		} else return false;
+	var filename;
+
+	// localPrefix is the folder our bundle should be in
+	var localPrefix = urlParse(publicPath || "/");
+	var urlObject = urlParse(url);
+
+	// publicPath has the hostname that is not the same as request url's, should fail
+	if(localPrefix.hostname !== null && urlObject.hostname !== null &&
+		   localPrefix.hostname !== urlObject.hostname) {
+		return false;
+	}
+
+	// publicPath is not in url, so it should fail
+	if(publicPath && localPrefix.hostname === urlObject.hostname && !new RegExp('^' + publicPath).test(url)) {
```

Comment:
I think `lastIndexOf`, see [this question](https://stackoverflow.com/questions/646628/how-to-check-if-a-string-startswith-another-string#answer-4579228).

Yes, the example you've given is expected to be `false`, because `publicPath` is not in the url. Would be nice if you add your example to the tests.